shopping-cart-room/src/androidTest/java/com/handstandsam/shoppingapp/cart/RoomShoppingCartTest.kt

```diff
@@ -0,0 +1,82 @@
+package com.handstandsam.shoppingapp.cart
+
+import android.os.Handler
+import android.os.Looper
+import com.handstandsam.shoppingapp.models.Item
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Test
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+/**
+ * How to test [Room] https://developer.android.com/training/data-storage/room/testing-db
+ */
+class RoomShoppingCartTest {
+
+    private lateinit var testDelegate: RoomShoppingCartTestDelegate
+
+    @Before
+    fun setUp() {
+        //Using a latch is a hack to force the cart observer to register on the Main thread, but wait until it's done.
+        val latch = CountDownLatch(1)
+        Handler(Looper.getMainLooper()).post {
+            testDelegate = RoomShoppingCartTestDelegate()
+            latch.countDown()
+        }
+        latch.await(1, TimeUnit.SECONDS)
```

Comment:
Using a `CountDownLatch` is a hack to make sure that this code runs synchronously, but also initializes the Room `LiveData` observer on the main thread.  This works fine in my `prod` code, but in the test, I have to force it.